packages/wpcom-checkout/src/checkout-line-items.tsx

```diff
@@ -812,11 +818,63 @@ export function LineItemSublabelAndPrice( { product }: { product: ResponseCartPr
 			<>
 				<DefaultLineItemSublabel product={ product } />:{ ' ' }
 				{ translate( '%(price)s per three years', { args: { price } } ) }
+				<LineItemExpiryDates product={ product } />
 			</>
 		);
 	}
 
-	return <DefaultLineItemSublabel product={ product } />;
+	return (
+		<>
+			<DefaultLineItemSublabel product={ product } />
+			<LineItemExpiryDates product={ product } />
+		</>
+	);
+}
+
+function LineItemExpiryDates( { product }: { product: ResponseCartProduct } ) {
+	const translate = useTranslate();
+	const expiryDate = product.subscription_current_expiry_date
+		? formatDate( product.subscription_current_expiry_date )
+		: undefined;
+	const postRenewExpiry = product.subscription_post_purchase_expiry_date
+		? formatDate( product.subscription_post_purchase_expiry_date )
+		: undefined;
+
+	// We only currently show the expiry date for renewals, although it is
```

Comment:
I added a similar comment in code-D165321#3090888, but I wonder if this should be included for more than just renewals. There are many upgrade paths (plan term upgrades, adding new mailboxes), which can result in confusion about the expiration date. I think it might be worth considering including the new expiration date anytime there is a pre-existing subscription that is affecting the new purchase price or billing term.